underfs/oss/src/main/java/alluxio/underfs/oss/OSSUnderFileSystem.java

```diff
@@ -96,7 +103,18 @@ public static OSSUnderFileSystem createInstance(AlluxioURI uri, UnderFileSystemC
   protected OSSUnderFileSystem(AlluxioURI uri, OSS ossClient, String bucketName,
       UnderFileSystemConfiguration conf) {
     super(uri, conf);
-    mClient = ossClient;
+
+    mStsEnabled = conf.getBoolean(PropertyKey.UNDERFS_OSS_STS_ENABLED);
+    if (mStsEnabled) {
+      try {
+        mClientProvider = new StsOssClientProvider(conf);
+      } catch (IOException e) {
+        LOG.error("init sts client provider failed!", e);
+        throw new ServiceException(e);
+      }
+    }
+    mClient = mStsEnabled ? mClientProvider.getOSSClient() : ossClient;
```

Comment:
Why not initialize the ossClient that will use STS in the construct method?

Reply:
Because createInstance is a static method, I can't init and assign StsOssClientProvider in a static method.